### PR TITLE
doc: provide board porting guidelines

### PR DESCRIPTION
**Pull request by erwango**
Provide board porting guidelines to help achieving a coherent
board library to ease work for people developping application
not board specific and define clear objective to help board
porting reviews.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

### CODE REVIEW

File: doc/porting/board_porting.rst [hunk 1]
```diff
@@ -120,3 +120,25 @@ Board
 A board implements an SoC with all its features, together with peripherals
 available on the board that differentiates the board with additional interfaces
 and features not available in the SoC.
+
+Default board configuration
+***************************
+
+When porting a board, you must provide its default configuration. This default
```
Comment (carlescufi) [inline doc/porting/board_porting.rst]: Should this be "when porting to a board"?

File: doc/porting/board_porting.rst [hunk 2]
```diff
@@ -120,3 +120,25 @@ Board
...
+configuration could be overridden, or otherwise will be the reference when an
```
Comment (carlescufi) [inline doc/porting/board_porting.rst]: "the reference one"

File: doc/porting/board_porting.rst [hunk 3]
```diff
@@ -120,3 +120,25 @@ Board
...
+application, test or sample is compiled for this board.
+
+In order to provide coherency accross the various boards and ease the work of
```
Comment (carlescufi) [inline doc/porting/board_porting.rst]: consistency better than coherency?

File: doc/porting/board_porting.rst [hunk 4]
```diff
@@ -120,3 +120,25 @@ Board
...
+users providing applications that are not board specific, following guidelines
```
Comment (carlescufi) [inline doc/porting/board_porting.rst]: the following guidelines

File: doc/porting/board_porting.rst [hunk 5]
```diff
@@ -120,3 +120,25 @@ Board
...
+should be followed when porting a board:
+
+- Provide pins and drivers configuration that matches the board valuable
```
Comment (carlescufi) [inline doc/porting/board_porting.rst]: Provide pin and driver configuration that matches...

File: doc/porting/board_porting.rst [hunk 6]
```diff
@@ -120,3 +120,25 @@ Board
...
+components, such as sensors, button, leds or communication i/f such as USB,
+Ethernet connector or Bluetooth/WiFi chip
+- When a well-known connector (arduino, 96board, ....) is present,
```
Comment (carlescufi) [inline doc/porting/board_porting.rst]: s/..../etc

File: doc/porting/board_porting.rst [hunk 7]
```diff
@@ -120,3 +120,25 @@ Board
...
+configure pins to fit this connector
+- Then configure coponents that enable the use of these pins (For instance
```
Comment (carlescufi) [inline doc/porting/board_porting.rst]: s/Then//

File: doc/porting/board_porting.rst [hunk 8]
```diff
@@ -120,3 +120,25 @@ Board
...
+configure SPI instance for arduino SPI)
+- Configure an output for console
```
Comment (carlescufi) [inline doc/porting/board_porting.rst]: the console

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: doc/porting/board_porting.rst (hunk 1 above)
Comment (dbkinder) [reply doc/porting/board_porting.rst]: Or

    When porting to a board, you must provide the board's default
    configuration, used (unless overridden) when an application, test,
    or sample is compiled for this board.

File: doc/porting/board_porting.rst (hunk 3 above)
Comment (dbkinder) [reply doc/porting/board_porting.rst]: also fix misspelling of ``across``

File: doc/porting/board_porting.rst [hunk 9]
```diff
@@ -120,3 +120,25 @@ Board
...
+Default board configuration
+***************************
+
+When porting a board, you must provide its default configuration. This default
+configuration could be overridden, or otherwise will be the reference when an
+application, test or sample is compiled for this board.
+
+In order to provide coherency accross the various boards and ease the work of
+users providing applications that are not board specific, following guidelines
+should be followed when porting a board:
```
Comment (dbkinder) [inline doc/porting/board_porting.rst]: ``porting to a board``

File: doc/porting/board_porting.rst [hunk 10]
```diff
@@ -120,3 +120,25 @@ Board
...
+
+- Provide pins and drivers configuration that matches the board valuable
+components, such as sensors, button, leds or communication i/f such as USB,
```
Comment (dbkinder) [inline doc/porting/board_porting.rst]: Make sure the second line of a list item is indented properly
    
    - Provide pin and driver configurations that match the board's
      valuable components such as sensors, buttons, or LEDs, and communication
      interfaces such as USB, Ethernet connector, or Bluetooth/WiFi chip.

File: doc/porting/board_porting.rst (hunk 6 above)
Comment (dbkinder) [reply doc/porting/board_porting.rst]: - When a well-known connector is present (such as used on an Arduino or 96board),
      configure pins to fit this connector.

File: doc/porting/board_porting.rst (hunk 7 above)
Comment (dbkinder) [reply doc/porting/board_porting.rst]: - Configure components that enable the use of these pins, such as
      configuring and SPI instance for Arduino SPI.

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/porting/board_porting.rst (hunk 1 above)
Comment (mbolivar) [reply doc/porting/board_porting.rst]: This is verbose. Why not something like:

```
When porting Zephyr to a board, you must provide the board's default
Kconfig configuration, which is used in application builds unless explicitly
overridden.
```

File: doc/porting/board_porting.rst (hunk 5 above)
Comment (mbolivar) [reply doc/porting/board_porting.rst]: also "the board's", not "the board"

File: doc/porting/board_porting.rst (hunk 7 above)
Comment (mbolivar) [reply doc/porting/board_porting.rst]: note the "components" typo correction pointed out by @dbkinder

File: doc/porting/board_porting.rst (hunk 1 above)
Comment (erwango) [reply doc/porting/board_porting.rst]: I'll take this last one, omitting "Kconfig" as it is true for any kind of configuration (pinmux, dts, ..)

**Comment by erwango** [comment]
Thanks all for the reviews. Text updated

**Review by carlescufi — APPROVED** [review]

File: doc/porting/board_porting.rst [hunk 11]
```diff
@@ -120,3 +120,30 @@ Board
...
+Default board configuration
+***************************
+
+When porting Zephyr to a board, you must provide the board's default
+Kconfig configuration, which is used in application builds unless explicitly
+overridden.
+
+In order to provide consistency across the various boards and ease the work of
+users providing applications that are not board specific, the following
+guidelines should be followed when porting a board:
```
Comment (mbolivar) [inline doc/porting/board_porting.rst]: Looks great! The only thing I would add to the end of this sentence is "when possible". Sometimes it is not possible to follow the guidelines, such as if the board does not have a network interface.

**Review by mbolivar — APPROVED** [review]
One minor idea, but I think this looks good.

**Review by dbkinder — APPROVED** [review]
thanks!

**Review by nashif — APPROVED** [review]

**Pull request merged**